File: assignments/01._file_format/Node/main.js
import fs from "fs";
import yaml from 'js-yaml';
import csv from 'csv-parser';
import xml2js from "xml2js";

// Jeg har her lavet de fleste om til JSON objekter
function csvHandler() {
    const res = [];
    fs.createReadStream("./files/me.csv")
        .pipe(csv())
        .on('data', (data) => res.push(data))
        .on('end', () => {
            console.log("CSV output: ", res[0]);
            return res[0];
        });
};
function txtHandler() {
    fs.readFile("./files/me.txt", "utf-8", (err, txt) => {
        txt = txt.split("\r\n");
        console.log("TXT output: ", txt);
    })
};

function xmlHandler() {
    fs.readFile('./files/me.xml', function (err, data) {
        const xmlData = data.toString();

        xml2js.parseString(xmlData, (err, result) => {
            if (err) {
                console.error(err);
            } else {
                const jsonData = JSON.stringify(result, null, 4);
                console.log("XML output: ", jsonData);
            }
        });
    });
};

function jsonHandler() {
    fs.readFile("./files/me.json", "utf-8", (err, json) => {
        json = JSON.parse(json);
        console.log("JSON output: ", json);
    });
};

function yamlHandler() {
    const data = yaml.load(fs.readFileSync('./files/me.yaml', 'utf8'));
    console.log("YAML output: ", data);
    return data
};

// Handlers
csvHandler();
txtHandler();
xmlHandler();
jsonHandler();
yamlHandler();

File: assignments/01._file_format/Python/main.py
import csv
import json
import yaml
import xmltodict

# Jeg har her lavet dem om til dicts

# CSV
def csvHandler():
    dic, count = {}, 0
    with open('./files/me.csv', 'r') as csv_file:
        reader = csv.reader(csv_file)
        for row in reader:
            if row[0]=="name":
                dic["headers"]=row
            else:
                dic["row{}".format(count)]=row
            count = count + 1
    print(dic)

# TXT
def txtHandler():
    dic, temp, count = {}, ["name","age","hobbies"],0
    with open("./files/me.txt","r") as txt_file:
        for row in txt_file:
            dic[temp[count]] = row.replace("\n","")
            count = count + 1
        print(dic)

# NODEJS
def jsonHandler():
    st = ""
    with open("./files/me.json","r") as json_file:
        for row in json_file:
            st = st + row
    jsonEnd = json.loads(st)
    print(jsonEnd)

# YAML
def yamlHandler():
    with open("./files/me.yaml","r") as yaml_file:
        print(yaml.safe_load(yaml_file))

# XML
def xmlHandler():
    with open("./files/me.xml","r") as xml_file:
        dic = xmltodict.parse(xml_file.read())
    print(dic)

# Handlers
csvHandler()
txtHandler()
jsonHandler()
yamlHandler()
xmlHandler()